core/lib/makeswift/components/customer-group-slot/customer-group-slot.makeswift.tsx

```diff
@@ -0,0 +1,123 @@
+import { Checkbox, Combobox, Group, List, Slot, Style } from '@makeswift/runtime/controls';
+import { ReactNode } from 'react';
+import useSWR from 'swr';
+
+import { CustomerGroupType, CustomersGroup } from '~/app/api/customer/groups/route';
+import { GetCustomerGroupResponse } from '~/client/queries/get-customer';
+import { runtime } from '~/lib/makeswift/runtime';
+
+async function fetchData(): Promise<CustomerGroupType> {
+  const response = await fetch('/api/customer/groups');
+
+  if (!response.ok) {
+    throw new Error(`HTTP error! status: ${response.status}`);
+  }
+
+  const data: unknown = await response.json();
+  const groups = CustomersGroup.parse(data);
+
+  return groups;
+}
+
+interface Props {
+  className: string;
+  slots?: Array<{ group?: string; slot: ReactNode }>;
+  simulateGroup: boolean;
+  simulatedGroup?: string;
+  noGroupSlot: ReactNode;
+}
+
+function CustomerGroupSlot({
+  className,
+  slots,
+  simulateGroup,
+  simulatedGroup = 'no-group',
+  noGroupSlot,
+}: Props) {
+  const allSlots = slots?.concat({ group: 'no-group', slot: noGroupSlot });
+
+  const { data, isLoading } = useSWR<GetCustomerGroupResponse>('/api/customer/group', async () => {
+    const response = await fetch('/api/customer/group');
+
+    // eslint-disable-next-line @typescript-eslint/no-unsafe-return
+    return response.json();
```

Comment:
I think this response needs to be validated before parsing it as json. You can start with checking `status.ok`. I think you'll still need to surround with a try/catch as well in case the network request itself fails. Any idea what should happen if that network request fails?